docarray/array/mixins/embed.py

```diff
@@ -150,19 +165,49 @@ def _set_embeddings_onnx(
         embed_model,
         collate_fn: 'CollateFnType',
         device: str = 'cpu',
+        device_id: int = 0,
         batch_size: int = 256,
         *args,
         **kwargs,
     ):
         # embed_model is always an onnx.InferenceSession
-        if device != 'cpu':
+        provider_options = None
+        if device == 'cpu':
+            providers = ['CPUExecutionProvider']
+        elif device == 'cuda':
             import onnxruntime as ort
 
             support_device = ort.get_device()
             if device.lower().strip() != support_device.lower().strip():
                 warnings.warn(
                     f'Your installed `onnxruntime` supports `{support_device}`, but you give {device}'
                 )
+            providers = ['CUDAExecutionProvider']
+            provider_options = [
```

Comment:
why don't we let the provider options totally free?

Reply:
can you elaborate? what do you mean by free?

Reply:
to have `provider_options` in `kwargs`

Reply:
We could add `providers` and the `provider_options` as optional kwargs. In the case `providers` is set, this one is used and otherwise, `providers` and `providers_options` are initialized like Bo has it implemented in this function.